openstack/containerinfra/v1/nodegroups/doc.go

```diff
@@ -0,0 +1,68 @@
+/*
+Package nodegroups provides methods for interacting with the Magnum node group API.
+
+All node group actions must be performed on a specific cluster,
+so the cluster UUID/name is required as a parameter in each method.
+
+
+Create a client to use:
+
+    opts, err := openstack.AuthOptionsFromEnv()
+    if err != nil {
+        panic(err)
+    }
+
+    provider, err := openstack.AuthenticatedClient(opts)
+    if err != nil {
+        panic(err)
+    }
+
+    client, err := openstack.NewContainerInfraV1(provider, gophercloud.EndpointOpts{Region: os.Getenv("OS_REGION_NAME")})
+    if err != nil {
+        panic(err)
+    }
+
+    client.Microversion = "1.9"
+
+
+Example of Getting a node group:
+
+    ng, err := nodegroups.Get(client, clusterUUID, nodeGroupUUID).Extract()
+    if err != nil {
+    	panic(err)
+    }
+    fmt.Printf("%#v\n", ng)
+
+
+Example of Listing node groups:
+
+    // Options:
+    // Marker:  pagination marker for large data sets. (UUID field from node group).
+    // Limit:   maximum number of resources to return in a single result.
+    // SortKey: column to sort results by. Default: id.
+    // SortDir: direction to sort. "asc" or "desc". Default: asc.
+    // Role:    list all nodegroups with the specified role.
+
+    listOpts := nodegroup.ListOpts{
+    	Role: "worker",
+    }
+
+    allPages, err := nodegroups.List(client, clusterUUID, listOpts).AllPages()
+    if err != nil {
+    	panic(err)
+    }
+
+    ngs, err := nodegroups.ExtractNodeGroups(allPages)
+    if err != nil {
+    	panic(err)
+    }
+
+    for _, ng := range ngs {
+    	fmt.Printf("%#v\n", ng)
+    }
+
+    // When listing node groups, only these fields are returned
+    // "uuid", "name", "flavor_id", "node_count", "role",
+    // "is_default", "image_id", "status", "stack_id"
```

Comment:
nit: This can be removed, too. The result struct will be rendered in a readable way. If you'd like add annotation comments to the relevant fields, that might work better:

```go
// UUID is the unique ID of the node group.
// This is returned both in List and Get.
UUID string `json:"uuid"`
```

Reply:
I've moved this information to the documentation of `List` itself. I think it's more readable that way than spreading it out over multiple lines of the NodeGroup struct.

https://github.com/gophercloud/gophercloud/pull/1774/commits/f911976aad959fa0b3ba9984657b43f4af11e67f#diff-781e55e8261bfc534c6345810f539b08R55-R59